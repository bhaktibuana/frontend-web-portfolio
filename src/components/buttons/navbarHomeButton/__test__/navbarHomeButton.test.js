import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import NavbarHomeButton from "..";

afterEach(() => {
  cleanup();
});

test("should render button", () => {
  render(<NavbarHomeButton text="Home" scrollTo="home" />);
  const buttonElement = screen.getByTestId("navbarHomeButton");
  
  expect(buttonElement).toBeInTheDocument();
  expect(buttonElement).toHaveTextContent("Home");
});

test("matches snapshot", () => {
  const tree = renderer
    .create(<NavbarHomeButton text="Home" scrollTo="home" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
