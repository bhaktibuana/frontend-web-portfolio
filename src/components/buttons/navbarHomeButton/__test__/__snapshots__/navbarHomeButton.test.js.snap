// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  className="sc-beqWaB fbTEKw"
  color={
    Object {
      "bgDark": "#202020",
      "bgGrey": "#f1f5f9",
      "bgLight": "#ffffff",
      "primary": "#14b8a6",
      "secondary": "#1e293b",
      "tertiary": "#f1f5f9",
      "textDark": "#ffffff",
      "textLight": "#202020",
    }
  }
  data-testid="navbarHomeButton"
  onClick={[Function]}
  style={Object {}}
>
  Home
</a>
`;
